### DISCUSSION TITLE
Why can't two foreign keys refer the same primary key of a table ?

### DISCUSSION THREAD

**Discussion by parseshyam** [prisma/prisma#2410] [Q&A] [answered]
```
model User {
  id             Int             @default(autoincrement()) @id
 ...
 .
 .
 .
 ...
  followRequests FollowRequest[]
}

model FollowRequest {
  id        Int               @default(autoincrement()) @id
  createdAt DateTime          @default(now())
  updatedAt DateTime          @default(now())
  status    followingApproval @default(PENDING)
  userId    Int
  followId  Int
  user      User              @relation(fields: [userId, followId], references: [id])  
}

enum followingApproval {
  PENDING
  APPROVED
  REJECTED
}
```

 @relation(fields: ***[userId, followId]***, references: [id])  ***throws unexpected error!***
***Why can't two foreign keys refers the same primary key of a table ?***

**Comment by Akxe** [comment]
This is really meant for stack overflow. You solve with 2 separate `@relation`

```
model FollowRequest {
  id        Int               @default(autoincrement()) @id
  createdAt DateTime          @default(now())
  updatedAt DateTime          @default(now())
  status    followingApproval @default(PENDING)
  userId    Int
  followId  Int
  user      User              @relation(name: "name1", fields: [userId], references: [id])
  follow      User              @relation(name: "name2", fields: [followId], references: [id])
}```

**Comment by parseshyam** [comment]
Thanks for replying,
by above approach,
got following error 

Error validating model "FollowRequest": Ambiguous relation detected. The fields `user` and `follow` in model `FollowRequest` both refer to `User`. Please provide different relation names for them by adding `@relation(<name>).
```

**Comment by Akxe** [comment]
Updated original answer

**Comment by parseshyam** [comment]
> Updated original answer

Now getting following error.

Error validating model "FollowRequest": Automatic related field generation would cause a naming conflict. Please add an explicit opposite relation field.

**Comment by Akxe** [comment]
Try StackOverflow then. This is really for bug or feature request. I know this works, but I am not sure how to help, I am new to this.

**Reply by janpio** [reply]
(We can actually convert all issues to discussions here on Github, which is a bit like StackOverflow, so no harm done <3 I did that to the issue now.)

**Reply by Akxe** [reply]
Awesome! :D

**Comment by parseshyam** [comment, marked as answer]
```
model User {
  id             Int             @default(autoincrement()) @id
 ...
 .
 .
 .
 ...
  followRequests FollowRequest[]
************************** missed the following *************************

  user           FollowRequest     @relation("name1")
  follow         FollowRequest     @relation("name2")

**************************************************************************

}

model FollowRequest {
  id        Int               @default(autoincrement()) @id
  createdAt DateTime          @default(now())
  updatedAt DateTime          @default(now())
  status    followingApproval @default(PENDING)
  userId    Int
  followId  Int
  user      User               @relation(name: "name1", fields: [userId], references: [id])
  follow    User               @relation(name: "name2", fields: [followId], references: [id])}

enum followingApproval {
  PENDING
  APPROVED
  REJECTED
}
```

@Akxe  Figured out above is the way to achieve the same. 
Since prisma automatic related field generation would cause a naming conflict. 
hence we need to explicit define the relation name in opposite relation field.

**Comment by ludralph** [comment]
Hi @parseshyam,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
